patches/minecraft/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -649,11 +649,20 @@
              if (!p_21065_.f_46443_ && pair.getFirst() != null && p_21065_.f_46441_.m_188501_() < pair.getSecond()) {
                 p_21066_.m_7292_(new MobEffectInstance(pair.getFirst()));
              }
-@@ -3234,6 +_,64 @@
+@@ -3234,6 +_,73 @@
        this.m_21166_(p_21191_ == InteractionHand.MAIN_HAND ? EquipmentSlot.MAINHAND : EquipmentSlot.OFFHAND);
     }
  
 +   /* ==== FORGE START ==== */
++   /**
++    * Tracks the attack strength of the player as set within {@link net.minecraft.world.entity.player.Player#attack(Entity)} for outside usage
```

Comment:
This Javadoc needs to mention by name the series of events for which this value is considered valid, which are
* LivingAttackEvent / AttackEntityEvent
* LivingHurtEvent
* LivingDamageEvent
* LivingDeathEvent
* LivingDropsEvent

Reply:
I agree with everything here except that I don't think the value would be valid in AttackEntityEvent considering the cached value is set *after* that event is called. However, you can just use `.getAttackStrengthScale()` in AttackEntityEvent because that does have a valid value at the time that event runs.